### DISCUSSION TITLE
different base_url

### DISCUSSION THREAD

**Discussion by alvisebacco** [CrowdStrike/falconpy#139] [Q&A] [answered]
In _service_class.py, class ServiceClass has hardcoded base_url: str = "https://api.crowdstrike.com" in the constructor object; but should happen that some tennant has https://api.us-2.crowdstrike.com.

**Comment by jshcodes** [comment, marked as answer]
The following __init__ function definition defines the default base_url value (which is https://api.crowdstrike.com) when an instance of a Service Class is created. 
```python
def __init__(self: object, access_token: str = None, auth_object: object = None, creds: dict = None,
             base_url: str = "https://api.crowdstrike.com", ssl_verify: bool = True) -> object:
```
This is to allow users to create an instance of a Service Class without having to specify a base_url value.
```python
from falconpy.cloud_connect_aws import Cloud_Connect_AWS as FalconAWS
falcon = FalconAWS(creds={"client_id": client_id, 
                          "client_secret": client_secret})
result = falcon.QueryAWSAccounts()
```
In order to specify a different base_url, you will need to pass that value when you create the instance of the class.
```python
from falconpy.cloud_connect_aws import Cloud_Connect_AWS as FalconAWS
falcon = FalconAWS(creds={"client_id": client_id, 
                          "client_secret": client_secret
                         },
                          base_url="https://api.us-2.crowdstrike.com")
result = falcon.QueryAWSAccounts()
```

Are you having issues when creating an instance of a Service Class using a different base_url value?

**Comment by alvisebacco** [comment]
Nice! Thx!
